code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -80,16 +80,94 @@
 	desc = "A sturdy shotgun with a longer magazine and a fixed tactical stock designed for non-lethal riot control."
 	icon_state = "riotshotgun"
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/riot
-	sawn_desc = "Come with me if you want to live."
+	var/style = 1
 
 /obj/item/weapon/gun/projectile/shotgun/riot/attackby(obj/item/A, mob/user, params)
 	..()
-	if(istype(A, /obj/item/weapon/circular_saw) || istype(A, /obj/item/weapon/gun/energy/plasmacutter))
-		sawoff(user)
-	if(istype(A, /obj/item/weapon/melee/energy))
-		var/obj/item/weapon/melee/energy/W = A
-		if(W.active)
-			sawoff(user)
+	if(istype(A, /obj/item/device/riot_upgrade))
+		var/obj/item/device/riot_upgrade/C = A
+		if(C.style == style)
+			to_chat(user, "<span class='notice'>This [src] is already set up in this style!</span>")
+			return
+		if(istype(src.loc, /obj/item/weapon/storage))
+			to_chat(user, "<span class='info'>How do you plan to modify [src] while it's in a bag.</span>")
+			return
+		if(magazine.ammo_count() || chambered)

```

Comment:
I'm not sure how I feel about this - it's a hilarious "fuck you" to the face of anyone who forgets to unload the gun before customizing it, which isn't consistent with the detective's revolver

Reply:
You may  want to review the detective's revolver code, as it does use afterattack too.

https://github.com/ParadiseSS13/Paradise/blob/c1c4b08807c0eb23e7e2c89ed50310f4c7551f88/code/modules/projectiles/guns/projectile/revolver.dm#L121
https://github.com/ParadiseSS13/Paradise/blob/c1c4b08807c0eb23e7e2c89ed50310f4c7551f88/code/modules/projectiles/guns/projectile/revolver.dm#L134